extensions/resteasy-reactive/quarkus-resteasy-reactive/runtime/src/main/java/io/quarkus/resteasy/reactive/server/runtime/websocket/VertxWebSocketParamExtractor.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.resteasy.reactive.server.runtime.websocket;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.core.parameters.ParameterExtractor;
+
+import io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.ServerWebSocket;
+
+//TODO: do we actually want vert.x websockets?
+//they are not our primary API but they are easy to support
+public class VertxWebSocketParamExtractor implements ParameterExtractor {
```

Comment:
So how about a `@HandlerFor(type = ServerWebSocket.class)` annotation? Or we make it implement `ParameterExtractor<ServerWebSocket>` but I was thinking with an annotation we can make extractors pluggable based on annotations too in the future: `@HandlerFor(annotatedWith = FroMage.class)` which would support parameters annotated with `@FroMage`.

I think this could be simpler than `MethodScannerBuildItem` and its context.

Reply:
That would be fine for a user facing API, but we still need the programmatic version to avoid having to index Quarkus itself.

Reply:
True.

Reply:
I really don't want RR to have to self-index, so I am very much against making an annotation based approach the only way to do this. If we think this is useful for end uses I think building an annotation based API over the programatic impl would be the best way to handle this.

Reply:
Sure, it can wait.